test-e2e/har/har.test.ts

```diff
@@ -0,0 +1,193 @@
+import 'mocha';
+import * as assert from 'assert';
+import { parseHttpArchiveText, compareEntries, isEquivalent } from './har'
+import * as fs from 'fs';
+import * as path from 'path';
+
+describe('experiment', () => {
+  it('should only care about attributes a and b', () => {
+    let model = {
+      a: {
+        b: null
+      },
+      c: null
+    }
+
+    let o1 = {
+      a: {
+        b: 2,
+        c: 3
+      },
+      c: 3,
+      d: 4
+    };
+
+    console.log(`typeof b: ${typeof(model.a.b)}`)
+
+    function pick<T>(model: T, obj: any): T {
+      return Object.keys(model).reduce((res, key) => {
+        if (!(key in obj)) return res;
+        if (model[key] === null) {
+          res[key] = obj[key]; 
+          return res;
+        }
+        res[key] = pick(model[key], obj[key]);
+        return res;
+      }, {} as T)
+    }
+    console.log(pick(model, o1))
```

Comment:
What is this function doing? It looks like it strips the keys from a bigger object to get two different objects in the same shape?

If so, there might be easier ways to do it. I suggest taking a look at lodash's documentation, or going for an easier to read approach

Reply:
While the comment was addressed to accidentally duplicated code that's since been deleted, it is still relevant to the implementation here:
https://github.com/segmentio/analytics.js-core/blob/6011643fb6e2f1c46c17096cfb19ad40f751d50b/test-e2e/har/har.ts#L27-L37

Correct, the `pick` function picks properties from `obj`, which could be anything resulting from `JSON.parse()`. What it picks is defined by `model` which is typed. It is just a depth first search. 

W.r.t. other ways to do this: if we were using `lodash.pick`, then we would be defining what we want to pick using a list of paths defined with dot syntax, and the drawback is losing typing and forcing us to maintain the interface in two different forms (one in typescript interface, the other being a list of dot notations). 

For this reason, I am going to keep the current implementation because I haven't found a more readable / maintainable way to do this. Please reopen if necessary.